room-battle.js

```diff
@@ -291,20 +307,32 @@ class BattleTimer {
 			if (isConnected === this.connected[slotNum]) continue;
 
 			if (!isConnected) {
-				// player has disconnected: don't wait longer than 6 ticks (1 minute)
+				// player has disconnected
 				this.connected[slotNum] = false;
-				if (this.settings.dcTimer) {
-					this.dcTicksLeft[slotNum] = DISCONNECTION_TICKS;
-				} else {
-					// arbitrary large number
-					this.dcTicksLeft[slotNum] = DISCONNECTION_TICKS * 10;
+				if (!Config.dcTimerBank) {
+					// don't wait longer than 6 ticks (1 minute)
+					if (this.settings.dcTimer) {
+						this.dcTicksLeft[slotNum] = DISCONNECTION_TICKS;
+					} else {
+						// arbitrary large number
+						this.dcTicksLeft[slotNum] = DISCONNECTION_TICKS * 10;
+					}
 				}
+
 				if (this.timerRequesters.size) {
+					let msg = `!`;
+
 					if (this.settings.dcTimer) {
-						this.battle.room.add(`|inactive|${this.battle.playerNames[slotNum]} disconnected and has a minute to reconnect!`).update();
-					} else {
-						this.battle.room.add(`|inactive|${this.battle.playerNames[slotNum]} disconnected!`).update();
+						msg = ` and has a minute to reconnect!`;
+					}
+					if (Config.dcTimerBank) {
+						if (this.dcTicksLeft[slotNum] > 0) {
+							msg = ` and is now using disconnection time!`;
```

Comment:
This should probably say how much dc time they have left, considering there's no countdown until 40 seconds.

Reply:
Oh wait, it's every 30 seconds _or_ every tick under 40 seconds. Still, though. This should probably say how much dc time they have left.

Reply:
Actually there is
`if (dcTicksLeft <= 4 || dcTicksLeft % 3 === 0) {` is used for the DC bank.